app/uk/gov/hmrc/tariffclassificationfrontend/connector/BindingTariffClassificationConnector.scala

```diff
@@ -59,8 +59,8 @@ class BindingTariffClassificationConnector @Inject()(configuration: AppConfig, c
     client.PUT[Case, Case](url = url, body = c)
   }
 
-  def createEvent(c: Case, e: NewEventRequest)(implicit hc: HeaderCarrier): Future[Event] = {
-    val url = s"${configuration.bindingTariffClassificationUrl}/cases/${c.reference}/events"
+  def createEvent(reference: String, e: NewEventRequest)(implicit hc: HeaderCarrier): Future[Event] = {
```

Comment:
Personally, I prefer us to always pass full objects around e.g. Case rather than just references. It forces us to validate that the Case exists first and means our method signatures dont risk becoming `doSomething(string, string, string, int)` etc

Reply:
I've changed the controller to check for case exists and passed the case down the stack